node-data/src/lib.rs

```diff
@@ -17,20 +17,102 @@ pub trait Serializable {
     where
         Self: Sized;
 
+    /// Write varint fields
+    fn write_varint<W: Write>(w: &mut W, v: u64) -> io::Result<()> {
```

Comment:
As suggested, for any fixed-size fields (like seed, hash, signature)  write_varint should be substituted with non-length-prefixed encoding.

Reply:
Agree, once dusk-blockchain is deprecated we can easily move to more adeguate encoding